puppet/zulip/manifests/postgresql_backups.pp

```diff
@@ -4,19 +4,44 @@
   include zulip::postgresql_common
 
   $wal_g_version = $zulip::common::versions['wal-g']['version']
+  $wal_g_commit_hash = $zulip::common::versions['wal-g']['sha1']
+  $wal_g_binary_hash = $zulip::common::versions['wal-g']['sha256'][$::os['architecture']]
   $bin = "/srv/zulip-wal-g-${wal_g_version}"
-  $package = "wal-g-pg-ubuntu-20.04-${zulip::common::goarch}"
 
-  # This tarball contains only a single file
-  zulip::external_dep { 'wal-g':
-    version        => $wal_g_version,
-    url            => "https://github.com/wal-g/wal-g/releases/download/v${wal_g_version}/${package}.tar.gz",
-    tarball_prefix => $package,
+  if $wal_g_binary_hash != undef {
+    # We have a binary for this arch
+    $package = "wal-g-pg-ubuntu-20.04-${zulip::common::goarch}"
+    # This tarball contains only a single file, which is extracted as $bin
+    zulip::external_dep { 'wal-g':
+      version        => $wal_g_version,
+      url            => "https://github.com/wal-g/wal-g/releases/download/v${wal_g_version}/${package}.tar.gz",
+      tarball_prefix => $package,
+      before         => File['/usr/local/bin/wal-g'],
+    }
+  } else {
+    include zulip::golang
+    $source_dir = "/srv/zulip-wal-g-src-${wal_g_version}"
+    exec { 'clone wal-g':
+      command => "git clone https://github.com/wal-g/wal-g.git --branch v${wal_g_version} ${source_dir}",
```

Comment:
Do we ever install `git`, or use it from anywhere other than `upgrade-zulip-from-git`?

An alternative would be to download the source [tarball](https://github.com/wal-g/wal-g/archive/refs/tags/v1.1.3-rc-with-build.tar.gz) and check its SHA-256.

Reply:
As I also documented in the commit message, those also don't suffice because the build requires submodules, and the metadata for those isn't in the auto-generated tarballs.

I'll make that note into a comment as well, in case others have the same thought.

Reply:
I think we almost certainly have an indirect `git` dependency already, but I'll add an explicit one.

Reply:
We have an explicit `git` pacakge ensure in `puppet/zulip/manifests/profile/base.pp` and `puppet/zulip_ops/manifests/profile/base.pp`